src/eth.rs

```diff
@@ -78,7 +166,10 @@ impl VitalikProvider {
         deal_id: DealID,
         window_num: u64,
     ) -> Result<BlockNum> {
-        let contract = self.contract.lock().await;
+        if !self.has_contract() {
+            return Err(anyhow!("No contract available"));
```

Comment:
this isn't necessary the contract should just be initialized at the start and if it fails it fails

Reply:
My thinking was you might not need a contract for lots of chain specific view functions, so I thought it aided usability later on.